libs/utils/energy_model.py

```diff
@@ -897,6 +1000,39 @@ def read_idle_states(cpu, domain_level):
                    root_power_domain=root_pd,
                    freq_domains=freq_domains)
 
+    @classmethod
+    def from_target(cls, target):
+        """
+        Create an EnergyModel by reading a target filesystem
+
+        If present, load an EM provided via dt using from_sd_target, since these
+        devices make the full EM available via the sched domain in sysfs. If
+        there is no EM at this location, attempt to load the simplified EM
+        made available via dedicated sysfs files.
+
+        :param target: Devlib target object to read filesystem from. Must have
+                       cpufreq and cpuidle modules enabled.
+        :returns: Constructed EnergyModel object based on the parameters
+                  reported by the target.
+        """
+        def em_present_in_sd(target):
+            def sge_path(cpu, domain, group, field):
+                f = '/proc/sys/kernel/sched_domain/cpu{}/domain{}/group{}/energy/{}'
+                return f.format(cpu, domain, group, field)
+            # Read all the files we might need in one go, otherwise this will take
+            # ages.
+            sge_globs = ['/proc/sys/kernel/sched_domain/cpu**/domain**/group**/energy/cap_states',]
+            sge_file_values = read_multiple_oneline_files(target, sge_globs)
+
+            if not sge_file_values:
+                return False
+            return True
+
+        if em_present_in_sd(target):
+            return cls.from_sd_target(target)
```

Comment:
Could you add some logging in here (or at the head of either EM loading methods) specifying which EM is being loaded ? So far we only have this log in **env.py**:
``` python
self._log.info('Attempting to read energy model from target')
```

Being able to tell from the log which one was actually loaded would be great :)

Reply:
If you use the simplified loader, you get a message but I agree I should probably let you know which it thinks you have.